apps/wallet/src/ui/app/staking/stake/StakeForm.tsx

```diff
@@ -21,30 +22,45 @@ export type StakeFromProps = {
     submitError: string | null;
     coinBalance: bigint;
     coinType: string;
+    coinDecimals: number;
     epoch: string;
     onClearSubmitError: () => void;
 };
 
+function AvailableBalance({ amount }: { amount: bigint }) {
+    const [formatted, symbol] = useFormatCoin(amount, SUI_TYPE_ARG);
+    return (
+        <Text variant="bodySmall" color="steel" weight="medium">
+            Available - {+formatted > 0 ? formatted : 0} {symbol}
+        </Text>
+    );
+}
+
 function StakeForm({
     submitError,
     coinBalance,
     coinType,
     onClearSubmitError,
+    coinDecimals,
     epoch,
 }: StakeFromProps) {
-    const { setFieldValue } = useFormikContext<FormValues>();
+    const {
+        setFieldValue,
+        values: { amount },
+    } = useFormikContext<FormValues>();
 
     const onClearRef = useRef(onClearSubmitError);
     onClearRef.current = onClearSubmitError;
+    const gasBudget = BigInt(
+        coinType === SUI_TYPE_ARG ? DEFAULT_GAS_BUDGET_FOR_STAKE : 0
+    );
 
     const [gasBudgetEstimation] = useFormatCoin(
         DEFAULT_GAS_BUDGET_FOR_STAKE,
         SUI_TYPE_ARG
     );
 
-    const coinBalanceMinusGas =
-        coinBalance -
-        BigInt(coinType === SUI_TYPE_ARG ? DEFAULT_GAS_BUDGET_FOR_STAKE : 0);
+    const coinBalanceMinusGas = coinBalance - gasBudget;
```

Comment:
we can do something like
```suggestion
    const totalAvailableBalance = coinBalance - BigInt(coinType === SUI_TYPE_ARG ? DEFAULT_GAS_BUDGET_FOR_STAKE : 0);
```